modules/highgui/test/test_gui.cpp

```diff
@@ -48,9 +48,11 @@ inline void verify_size(const std::string &nm, const cv::Mat &img)
 {
     EXPECT_NO_THROW(imshow(nm, img));
     EXPECT_EQ(-1, waitKey(200));
+#ifndef HAVE_WAYLAND // Wayland backend is not supported getWindowImageRect().
     Rect rc;
     EXPECT_NO_THROW(rc = getWindowImageRect(nm));
     EXPECT_EQ(rc.size(), img.size());
+#endif // HAVE_WAYLAND
```

Comment:
I propose to use runtime check with `currentUIFramework` to support build with UI plugins. Also please add reference to ticket to the check comment.

Reply:
Thank you for your review, I fixed it.